test/test_ops.py

```diff
@@ -1102,296 +1101,381 @@ def test_bbox_convert_jit(self):
         )
 
         scripted_fn = torch.jit.script(ops.box_convert)
-        TOLERANCE = 1e-3
 
         box_xywh = ops.box_convert(box_tensor, in_fmt="xyxy", out_fmt="xywh")
         scripted_xywh = scripted_fn(box_tensor, "xyxy", "xywh")
-        torch.testing.assert_close(scripted_xywh, box_xywh, rtol=0.0, atol=TOLERANCE)
+        torch.testing.assert_close(scripted_xywh, box_xywh)
 
         box_cxcywh = ops.box_convert(box_tensor, in_fmt="xyxy", out_fmt="cxcywh")
         scripted_cxcywh = scripted_fn(box_tensor, "xyxy", "cxcywh")
-        torch.testing.assert_close(scripted_cxcywh, box_cxcywh, rtol=0.0, atol=TOLERANCE)
+        torch.testing.assert_close(scripted_cxcywh, box_cxcywh)
 
 
-class BoxTestBase(ABC):
-    @abstractmethod
-    def _target_fn(self) -> Tuple[bool, Callable]:
-        pass
+class TestBoxArea:
+    def area_check(self, box, expected, atol=1e-4):
+        out = ops.box_area(box)
+        torch.testing.assert_close(out, expected, rtol=0.0, check_dtype=False, atol=atol)
+
+    @pytest.mark.parametrize("dtype", [torch.int8, torch.int16, torch.int32, torch.int64])
+    def test_int_boxes(self, dtype):
+        box_tensor = torch.tensor([[0, 0, 100, 100], [0, 0, 0, 0]], dtype=dtype)
+        expected = torch.tensor([10000, 0], dtype=torch.int32)
+        self.area_check(box_tensor, expected)
+
+    @pytest.mark.parametrize("dtype", [torch.float32, torch.float64])
+    def test_float_boxes(self, dtype):
+        box_tensor = torch.tensor(
+            [
+                [285.3538, 185.5758, 1193.5110, 851.4551],
+                [285.1472, 188.7374, 1192.4984, 851.0669],
+                [279.2440, 197.9812, 1189.4746, 849.2019],
+            ],
+            dtype=dtype,
+        )
+        expected = torch.tensor([604723.0806, 600965.4666, 592761.0085], dtype=dtype)
+        self.area_check(box_tensor, expected)
+
+    def test_float16_box(self):
+        box_tensor = torch.tensor(
+            [[28.25, 18.625, 39.0, 48.5], [28.25, 48.75, 192.0, 51.0], [29.25, 18.0, 89.0, 49.0]],
+            dtype=torch.float16,
+        )
+        expected = torch.tensor([321.1562, 368.4375, 1852.2500], dtype=torch.float16)
+        self.area_check(box_tensor, expected, atol=0.3)
```

Comment:
Well sadly for small boxes and keeping both as float16. We need to loosen atol. I can't find some other workaround.

Reply:
It's hard to decipher the diff. What test does this correspond to in `main`? It looks like input and generated values were changed. Is this on purpose?

Reply:
It is on purpose. So that box_area after multiplication too stays in float16 input format.

Reply:
Important: The `atol=0.3` is extremely large and might lead to not being able to detect problems. Is that the minimum you can do? If yes then it's worth considering reviewing the values of the boxes so you can use smaller values like `0.001` which I believe was used previously.